rustls/src/quic.rs

```diff
@@ -182,45 +307,50 @@ pub(crate) fn write_hs(this: &mut ConnectionCommon, buf: &mut Vec<u8>) -> Option
         }
     }
 
-    let suite = this
-        .get_suite()
-        .and_then(|suite| suite.tls13())?;
     if let Some(secrets) = this.quic.hs_secrets.take() {
-        return Some(Keys::new(suite, this.is_client, &secrets));
+        return Some(KeyMaterial::Handshake {
+            keys: Keys::new(&secrets),
+        });
     }
 
-    if let Some(secrets) = this.quic.traffic_secrets.as_ref() {
+    if let Some(secrets) = this.quic.traffic_secrets.take() {
         if !this.quic.returned_traffic_keys {
             this.quic.returned_traffic_keys = true;
-            return Some(Keys::new(suite, this.is_client, secrets));
+            return Some(KeyMaterial::OneRtt {
+                keys: Keys::new(&secrets),
+                next: secrets.next_1rtt_secrets(),
+            });
         }
     }
 
     None
 }
 
-pub(crate) fn next_1rtt_keys(this: &mut ConnectionCommon) -> Option<PacketKeySet> {
-    let suite = this
-        .get_suite()
-        .and_then(|suite| suite.tls13())?;
-    let secrets = this.quic.traffic_secrets.as_ref()?;
-    let next = next_1rtt_secrets(suite.hkdf_algorithm, secrets);
-
-    let (local, remote) = next.local_remote(this.is_client);
-    let keys = PacketKeySet {
-        local: PacketKey::new(suite, local),
-        remote: PacketKey::new(suite, remote),
-    };
-
-    this.quic.traffic_secrets = Some(next);
-    Some(keys)
+/// Key material for use in QUIC packet spaces
+#[allow(clippy::large_enum_variant)]
+pub enum KeyMaterial {
```

Comment:
Usually we use the term "key material" to refer to the raw bytes of the key, which I think we're trying not to expose outside of Rustls, so `KeyMaterial` seems like a bad name.

Reply:
Got any suggestions?

Reply:
IIUC, `KeyMaterial` represents a notification that the keys used by the QUIC implementation need to be replaced. So maybe `TrafficKeyChange` or `NewTrafficKey`. I was thinking also `KeyUpdate` but that term has specific meaning in TLS.

Reply:
How about just `KeyChange`? The first set of keys is for handshake, not traffic.

Reply:
Renamed it as proposed.